src/main/java/hudson/plugins/git/util/BuildData.java

```diff
@@ -243,4 +243,22 @@ public String toString() {
                 ",buildsByBranchName="+buildsByBranchName+
                 ",lastBuild="+lastBuild+"]";
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (!(o instanceof BuildData)) {
+            return false;
+        } else {
+            BuildData otherBuildData = (BuildData) o;
+
+            if (otherBuildData.remoteUrls.equals(this.remoteUrls)
```

Comment:
> Would you be willing to consider

[Comparison](https://github.com/MarkEWaite/git-plugin/compare/master...master-PR388-BuildData-equals)

Other than the verbose implementation (Java 7+ `Objects` makes this a lot cleaner), seems fine to me, but the question is really for @abayer the original author.

> I don't see any annotation which hints that they won't be null, so I assume they can be null

I really do not know. From casual inspection it seems that `remoteUrls` and `buildsByBranchName` are intended to be `@Nonnull` while `lastBuild` is intended to be `@CheckForNull`. The fact the fields are `public` non-`final`, and set in some constructors but not others, is disturbing. Probably the class should be tightened up with `@AdaptField`.